bodhi/tests/server/functional/test_updates.py

```diff
@@ -2520,6 +2520,353 @@ def test_edit_testing_update_reset_karma_with_same_tester(self, publish, *args):
+
+        up = self.db.query(Update).filter_by(title=nvr).one()
+        self.assertFalse(up.has_negative_karma)
+
+    @mock.patch(**mock_valid_requirements)
+    @mock.patch('bodhi.server.notifications.publish')
+    def test_disable_autopush_for_critical_updates(self, publish, *args):
+        """Make sure that autopush is disabled if critical update receives any negative karma"""
+        user = User(name=u'bob')
+        self.db.add(user)
+        self.db.flush()
+
+        nvr = u'kernel-3.11.5-300.fc17'
+        args = self.get_update(nvr)
+        args['autokarma'] = True
+        resp = self.app.post_json('/updates/', args)
+        self.assertTrue(resp.json['critpath'])
+        self.assertEquals(resp.json['request'], 'testing')
+        publish.assert_called_with(topic='update.request.testing', msg=ANY)
+
+        upd = self.db.query(Update).filter_by(title=resp.json['title']).one()
+        upd.status = UpdateStatus.testing
+        upd.request = None
+        self.db.flush()
+
+        upd.comment(self.db, u'Failed to work', author=u'ralph', karma=-1)
+        upd.disable_autopush_for_critical_update()
+        up = self.db.query(Update).filter_by(title=resp.json['title']).one()
+        self.assertTrue(up.has_negative_karma)
+        self.assertEquals(up.karma, -1)
+        self.assertEquals(up.status, UpdateStatus.testing)
+        self.assertEquals(up.request, None)
+        self.assertEquals(up.autokarma, False)
+
+    @mock.patch(**mock_valid_requirements)
+    @mock.patch('bodhi.server.notifications.publish')
+    def test_autopush_critical_update_with_no_negative_karma(self, publish, *args):
+        """Autopush critical update when it has no negative karma"""
+        user = User(name=u'bob')
+        self.db.add(user)
+        self.db.flush()
+
+        nvr = u'kernel-3.11.5-300.fc17'
+        args = self.get_update(nvr)
+        args['autokarma'] = True
+        args['stable_karma'] = 2
+        args['unstable_karma'] = -2
+
+        resp = self.app.post_json('/updates/', args)
+        self.assertTrue(resp.json['critpath'])
+        self.assertEquals(resp.json['request'], 'testing')
+        publish.assert_called_with(topic='update.request.testing', msg=ANY)
+
+        upd = self.db.query(Update).filter_by(title=resp.json['title']).one()
+        upd.status = UpdateStatus.testing
+        self.db.flush()
+
+        upd.comment(self.db, u'LGTM', author=u'ralph', karma=1)
+        up = self.db.query(Update).filter_by(title=resp.json['title']).one()
+        self.assertFalse(up.has_negative_karma)
+
+        up.comment(self.db, u'LGTM', author=u'bowlofeggs', karma=1)
+        up = self.db.query(Update).filter_by(title=resp.json['title']).one()
+        self.assertFalse(up.has_negative_karma)
+        self.assertEquals(up.karma, 2)
+        self.assertEquals(up.autokarma, True)
+
+        up = self.db.query(Update).filter_by(title=resp.json['title']).one()
+        self.assertEquals(up.request, UpdateRequest.stable)
```

Comment:
Well we do not need to run the method `check_karma_thresholds`. 
I am attaching the screenshot of debugging for better clarification. 

![screenshot from 2016-09-15 18-30-09](https://cloud.githubusercontent.com/assets/13337039/18575619/6dba7a66-7bf7-11e6-98c9-25942b7d409e.png)